galsim/gsobject.py

```diff
@@ -1351,57 +1322,410 @@ def drawImage(self, image=None, nx=None, ny=None, bounds=None, scale=None, wcs=N
+                            necessary to result in an image with the correct Poisson shot
+                            noise for the object's flux.  For positive definite profiles, this
+                            is equivalent to `n_photons = flux`.  However, some profiles need
+                            more than this because some of the shot photons are negative
+                            (usually due to interpolants).  [default: 0]
+        @param rng          If provided, a random number generator to use for photon shooting,
+                            which may be any kind of BaseDeviate object.  If `rng` is None, one
+                            will be automatically created, using the time as a seed.
+                            [default: None]
+        @param max_extra_noise  If provided, the allowed extra noise in each pixel when photon
+                            shooting.  This is only relevant if `n_photons=0`, so the number of
+                            photons is being automatically calculated.  In that case, if the image
+                            noise is dominated by the sky background, then you can get away with
+                            using fewer shot photons than the full `n_photons = flux`.  Essentially
+                            each shot photon can have a `flux > 1`, which increases the noise in
+                            each pixel.  The `max_extra_noise` parameter specifies how much extra
+                            noise per pixel is allowed because of this approximation.  A typical
+                            value for this might be `max_extra_noise = sky_level / 100` where
+                            `sky_level` is the flux per pixel due to the sky.  Note that this uses
+                            a "variance" definition of noise, not a "sigma" definition.
+                            [default: 0.]
+        @param poisson_flux Whether to allow total object flux scaling to vary according to
+                            Poisson statistics for `n_photons` samples when photon shooting.
+                            [default: True, unless `n_photons` is given, in which case the default
+                            is False]
+        @param gain         The number of photons per ADU ("analog to digital units", the units of
+                            the numbers output from a CCD).  [default: 1.]
+
+        @returns The total flux of photons that landed inside the image bounds.
+        """
+        # Make sure the type of n_photons is correct and has a valid value:
+        if n_photons < 0.:
+            raise ValueError("Invalid n_photons < 0.")
+
+        if poisson_flux is None:
+            if n_photons == 0.: poisson_flux = True
+            else: poisson_flux = False
+
+        # Check that either n_photons is set to something or flux is set to something
+        if (n_photons == 0. and self.getFlux() == 1.
+            and area == 1. and exptime == 1.): # pragma: no cover
```

Comment:
Fixed.  And added test that would have caught this.  (And removed no cover)  9296ac1